### ISSUE TITLE
Possible confusion between dropout rate vs keep_prob rate

### ISSUE THREAD

**Issue by zeran4** [hunkim/DeepLearningZeroToAll#122]
# Problem
* `tf.layers.dropout` uses `dropout rate`
    * 0.7 means 70% will be dropped
    * [API doc](https://www.tensorflow.org/api_docs/python/tf/layers/dropout)
* `tf.contrib.layers.dropout` uses `keep_prob` instead
    * 0.7 means 70% will be kept
    * [API doc](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/dropout)

# Suggestion
* In [lab-11-4-mnist_cnn_layers.py](https://github.com/hunkim/DeepLearningZeroToAll/blob/27450fcc4f87e0b6a1e6fbbb50020e129570c3e4/lab-11-4-mnist_cnn_layers.py), rate should be changed to 0.3 from 0.7
* In [lab-11-5-mnist_cnn_ensemble_layers.py](https://github.com/hunkim/DeepLearningZeroToAll/blob/4abd353993b0f0af8cc1720834b0dc85b2b3c084/lab-11-5-mnist_cnn_ensemble_layers.py), rate should be changed to 0.3 from 0.7

**Comment by sxjscience** [comment]
@zeran4 I've tested these two scripts and in fact a keeping probability of 0.7 is correct. If we change the keeping probability to 0.3, too many nodes will be dropped and the performance will be quite bad.

You can refer to the implementation using MXNet https://github.com/hunkim/DeepLearningZeroToAll/blob/master/mxnet/mxlab-11-2-mnist_deep_cnn.py#L27

**Comment by kkweon** [comment]
@sxjscience 

@zeran4's issue is not about lowering the keep_prob to 0.3 from 0.7, but it's more like a confusing TensorFlow syntax.

In TensorFlow, there are two functions that perform the dropout

* `tf.layers.dropout`
    * If you look at the doc, it uses the "dropout" rate
    * it behaves like MXNet

* `tf.contrib.layers.dropout` (same as `tf.nn.dropout`)
    * It uses the "keep_prob" rate

By dropout rate, I mean the ratio of nodes to be dropped
By keep_prob, I mean the ratio nodes to be kept

dropout rate = 1 - keep_prob

So,

In lab-11-4-mnist_cnn_layers.py, the rate should be changed to 0.3 from 0.7
because it uses `tf.layers.dropout` and it uses the `dropout rate` not the `keep_prob`

**Comment by sxjscience** [comment]
I see. So it's not consistent with https://github.com/hunkim/DeepLearningZeroToAll/blob/master/lab-11-2-mnist_deep_cnn.py#L51

**Comment by chg0901** [comment]
In the file [DeepLearningZeroToAll/lab-10-5-mnist_nn_dropout.py](https://github.com/hunkim/DeepLearningZeroToAll/blob/master/lab-10-5-mnist_nn_dropout.py#L24), we have this problem too

**Comment by kkweon** [comment]
finally fixed with #205 #206
